lib/Dialect/Lattigo/IR/LattigoBGVOps.td

```diff
@@ -116,4 +117,47 @@ def Lattigo_BGVMulOp : Lattigo_BGVBinaryOp<"mul"> {
   }];
 }
 
+class Lattigo_BGVUnaryOp<string mnemonic> :
+        Lattigo_BGVOp<mnemonic> {
+  let arguments = (ins
+    Lattigo_BGVEvaluator:$evaluator,
+    Lattigo_RLWECiphertext:$input
+  );
+  let results = (outs Lattigo_RLWECiphertext:$output);
+}
+
+def Lattigo_BGVRelinearizeOp : Lattigo_BGVUnaryOp<"relinearize"> {
+  let summary = "Relinearize a ciphertext in the Lattigo BGV dialect";
+  let description = [{
+    This operation relinearizes a ciphertext value in the Lattigo BGV dialect.
+  }];
+}
+
+def Lattigo_BGVRescaleOp : Lattigo_BGVUnaryOp<"rescale"> {
+  let summary = "Rescale a ciphertext in the Lattigo BGV dialect";
+  let description = [{
+    This operation rescales a ciphertext value in the Lattigo BGV dialect.
+  }];
+}
+
+def Lattigo_BGVRotateColumnsOp : Lattigo_BGVOp<"rotate_columns"> {
+  let summary = "Rotate columns of a ciphertext in the Lattigo BGV dialect";
+  let description = [{
+    This operation rotates the columns of a ciphertext value in the Lattigo BGV dialect.
```

Comment:
The lattigo code comments don't= explain the difference between rows and columns. Do you know? I expect "columns" corresponds to slots in the standard sense, but what does "rows" mean? It might be worth documenting here, or putting a link to where it is explained.

Reply:
I do have a reference comment in SEAL

https://github.com/microsoft/SEAL/blob/119dc32e135cb89c1062076a69310d4413ebc824/native/examples/2_encoders.cpp#L37-L106

But I can not immediately refer to a paper to explain this. Will investigate later.